src/godot_fmod.cpp

```diff
@@ -377,14 +381,14 @@ bool Fmod::_is_dead(Node* node) {
     if (!node) {
         return true;
     }
-    return !godot::core_1_1_api->godot_is_instance_valid(node->_owner);
+    return node->get_owner() == nullptr;

```

Comment:
I *could* be wrong, but I don't think `is_instance_valid()` is merely a `nullptr` check. I think it's a bit more than just that, e.g. checking if a node has been marked for deletion as well and other edge cases like that. `is_instance_valid()` is a global function, so based on the other code I've seen in this PR, I suspect you would find it in GDExtensions as `UtilityFunctions::is_instance_valid(Object *obj)`.

Reply:
As discussed, this internally does a null ptr check. I'd be in favour of keeping your suggestion, though. It makes it more explicit.

Reply:
Actually, if you have the object pointer already, then evaluating it with `is_instance_valid()` would be less clear from the C++ side of things. `is_instance_valid()` accepts a `Variant`, not an `Object*` which means that in order to perform the operation, you implicitly cast from `Node*` to `Variant` (creating another object on the stack in the process) and then take longer to properly process the expression only to come to the same conclusion. If you were *starting out* with a `Variant`, it would make more sense, but with you already having a `Node*`, it just makes for harder-to-read C++ code. Like, people would wonder why you are choosing to use that method instead of just directly checking, believing that it would have been for a specific reason.

Reply:
The issue with checking if a node was a bit more complicated than that because we ran in many cases where the same memory adress was reused for a new Godot object so checking stuff like an owner wouldn't necessary return a nullptr.
Actually in the current master, we still have many cornercases of Node being dead but without our module noticing.
Normally this kind of issue should be fixed in Godot 4.0 because the new ObjectDB handle objects lifecycle differently now. But it still means that we have no call that ObjectDB directly or indirectly. 
I don't know if there is any function to do in the current GDExtension (I haven't checked the current API for the GDExtension for quite a time so I don't know about recent evolutions).